doc/dev_guide/writing_extensions.rst

```diff
@@ -230,26 +230,31 @@ Equivalently, to add a new component 2D:
 Creating and registering new widgets and toolkeys
 -------------------------------------------------
 
-Toolkeys are functions to which it is possible to associate widgets. Extension
-can declare new toolkeys and widgets. For example, the `hyperspy-gui-traitsui
-<https://github.com/hyperspy/hyperspy_gui_traitsui>`_ and
-`hyperspy-gui-ipywidgets
-<https://github.com/hyperspy/hyperspy_gui_ipywidgets>`_ provide widgets for
-toolkeys declared in HyperSpy.
+To generate GUIs of specific method and functions, HyperSpy use widgets and toolkeys:
+
+* *widgets* are ipywidgets or traitsui objects, which can generate GUIs,
+* *toolkeys* are functions to which it is possible to associate widgets to a signal 
+  method or to a module function.
+
+An extension can declare new toolkeys and widgets. For example, the 
+`hyperspy-gui-traitsui <https://github.com/hyperspy/hyperspy_gui_traitsui>`_ and
+`hyperspy-gui-ipywidgets <https://github.com/hyperspy/hyperspy_gui_ipywidgets>`_ 
+provide widgets for toolkeys declared in HyperSpy.
 
 Registering toolkeys
 ^^^^^^^^^^^^^^^^^^^^
+To register a new toolkey:
 
-Typically new toolkeys are declared using the
-:py:func:`hyperspy.ui_registry.add_gui_method` HyperSpy decorator.
-To register a new toolkey that you have declared in your package, add it to
-the ``hyperspy_extension.yaml`` file as in the following example:
+1. declare a new toolkey by adding the :py:func:`hyperspy.ui_registry.add_gui_method` 
```

Comment:
I would add "e.g." in front of ``add_gui_method`` because there are other ways to do it.

Reply:
Shall we also mention the `get_gui` function?

Reply:
I think that most developers will simply read the code to find it out, but do as you feel is best.